### PR TITLE
Add API for reading and polling

### PR DESCRIPTION
**Pull request by jjcarstens**
This is a big cleanup. It adds the reading and polling API.

Polling is done via `telemetry_poller` and emits telemetry events that can be acted on as needed (that's to @hez for the suggestion!)

### CODE REVIEW

**Comment by hez** [comment]
@jjcarstens this is amazing! going to give it a quick spin today! <3

File: mix.exs [hunk 1]
```diff
@@ -30,19 +35,21 @@ defmodule Dht.MixProject do
 
   defp deps do
     [
+      {:ex_doc, "~> 0.22", only: :docs},
       {:circuits_gpio, "~> 0.4"},
-      {:elixir_make, "~> 0.6"}
+      {:elixir_make, "~> 0.6"},
+      # {:telemetry_poller, "~> 0.5"},
+      # TODO: remove this once https://github.com/beam-telemetry/telemetry_poller/pull/47 merged
+      {:telemetry_poller, github: "jjcarstens/telemetry_poller"}
```
Comment (hez) [inline mix.exs:43]: Any way to use the stock telemetry poller, gets complicated if this requires a fork.


```
Dependencies have diverged:
* telemetry_poller (https://github.com/jjcarstens/telemetry_poller.git)
  different specs were given for the telemetry_poller app:

  > In deps/dht/mix.exs:
    {:telemetry_poller, [targets: :rpi3, env: :prod, git: "https://github.com/jjcarstens/telemetry_poller.git"]}

  > In ../home_hub/mix.exs:
    {:telemetry_poller, "~> 0.4", [env: :prod, repo: "hexpm", hex: "telemetry_poller"]}

  Ensure they match or specify one of the above in your deps and set "override: true"
** (Mix) Can't continue due to errors on dependencies
```

Comment (jjcarstens) [reply mix.exs:43]: Ya, that's the plan. My PR has been approved, but I'm just waiting for it to be merged and released
https://github.com/beam-telemetry/telemetry_poller/pull/47

Without it, this whole poller implementation is broken. So for testing, I'd say override the dep for now. Hopefully all goes through in the next couple of days

Comment (hez) [reply mix.exs:43]: Nice, that works as well!

Override: true is the route I went down for testing :D

**Comment by hez** [comment]
👍 Just gave it a test run, working like a charm

**Pull request merged**